pkg/graveler/settings/manager.go

```diff
@@ -71,20 +95,55 @@ func (m *Manager) Save(ctx context.Context, repositoryID graveler.RepositoryID,
 	if err != nil {
 		return err
 	}
-	err = m.blockAdapter.Put(ctx, block.ObjectPointer{
+
+	data := Settings{
+		Identifier: fmt.Sprintf(settingsUniqueRelativeKey, m.committedBlockStoragePrefix, key, xid.NewWithTime(time.Now())),
+	}
+	obj := block.ObjectPointer{
 		StorageNamespace: string(repo.StorageNamespace),
-		Identifier:       fmt.Sprintf(settingsRelativeKey, m.committedBlockStoragePrefix, key),
+		Identifier:       data.Identifier,
 		IdentifierType:   block.IdentifierTypeRelative,
-	}, int64(len(messageBytes)), bytes.NewReader(messageBytes), block.PutOpts{})
+	}
+	// Save to  block store
+	err = m.blockAdapter.Put(ctx, obj, int64(len(messageBytes)), bytes.NewReader(messageBytes), block.PutOpts{})
+	if err != nil {
+		return err
+	}
+
+	// Write key in KV Store
+	err = m.store.SetMsg(ctx, graveler.RepoPartition(&graveler.RepositoryRecord{RepositoryID: repositoryID, Repository: repo}), []byte(settingsPath(key)), &data)
```

Comment:
If succesful, you should delete the old object from the store, it could accumulate pretty fast..